### ISSUE TITLE
SPC gets stuck handling some mods

### ISSUE THREAD

**Issue by Griefed** [Griefed/ServerPackCreator#80]
* **What is the current behavior?**
ServerPackCreator gets stuck handling some mods both on versions 2.1.1 (and presumably any prior version) as well as 3.0.0-alpha.5. 

When working with version 2.1.1 there seems to be no workaround, whereas with 3.0.0-alpha.5 the workaround seems to be to disable the sideness-scanning.

The mod in question appears to be one with the ID `itemfilter` though seaching for said ID on CurseForge yields no result. As such, I need to name of the mod to run tests with. Idealy a modpack that contains said mod.

* **If the current behavior is a bug, please provide the steps to reproduce**
Imcomplete:
* Have a modpack containing the mod with the ID `itemfilter` (modname unknown as of the time of me writing this issue)
* Create a server pack using version 2.1.1
* Create a server pack using 3.0.0-alpha.5
    * Disable sideness-scanning (appears to be a workaround)
    * Enable sideness-scanning

* **What is the expected behavior?**
A server pack should be generated regardless of what mods are in the modpack.

* **Please tell us about your environment:**

- Your Operating System: Unknown, probably any
- Your Java version:  jre 1.8.0_291
- Your ServerPackCreator version: 2.1.1 and 3.0.0-alpha.5
- Your config file: Unknown
- Your latest log file: Unknown

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

Two users reported:
* 1.16.5 pack
* forge 36.2.0

**Comment by Griefed** [comment]
A user provided a modpack and mod with which it may be possible to replicate this issue:
* Modpack: https://www.curseforge.com/minecraft/modpacks/dungeons-and-dining-and-dragons-and-more/files/3474662
* Mod: https://www.curseforge.com/minecraft/mc-mods/snow-under-trees

**Comment by Griefed** [comment]
Turns out it is Spice of Life: Potato Edition. (https://www.curseforge.com/minecraft/mc-mods/spice-of-life-potato-edition)

The dependencies in it's mods.toml file are fudged up
```toml
# Dependencies are optional.
[[dependencies]]
    modId = "forge"
    mandatory = true
    versionRange = "[34,)"
    ordering = "NONE" # BEFORE or AFTER required if the relationship is not mandatory
    side = "BOTH" # BOTH, CLIENT or SERVER

[[dependencies]]
    modId = "minecraft"
    mandatory = true
    versionRange = "[1.16.3, )"
    ordering = "NONE"
    side = "BOTH"
```

They're missing their modID [[dependencies.this_is_where_the_mod_should_have_its_own_ID]]

**Comment by Griefed** [comment]
Disenchanting also as the wrong modId in its dependencies......

```toml
# The modid of the mod
modId="disenchanting" #mandatory
[...]
# A file name (in the root of the mod JAR) containing a logo for display
logoFile="disenchanter.png" #optional
[...]
# A dependency - use the . to indicate dependency for a specific modid. Dependencies are optional.
[[dependencies.disenchanter]] #optional
    # the modid of the dependency
    modId="forge" #mandatory
    # Does this dependency have to exist - if not, ordering below must be specified
    mandatory=true #mandatory
    # The version range of the dependency
    versionRange="[28,)" #mandatory
    # An ordering relationship for the dependency - BEFORE or AFTER required if the relationship is not mandatory
    ordering="NONE"
    # Side this dependency is applied on - BOTH, CLIENT or SERVER
    side="BOTH"
# Here's another dependency
[[dependencies.disenchanter]]
    modId="minecraft"
    mandatory=true
    versionRange="[35.1.28]"
    ordering="NONE"
    side="BOTH"
```

Sigh

**Comment by Griefed** [comment]
Fix will be available in the next alpha release.

**Comment by Griefed** [comment]
Available as of [3.0.0-alpha.6](https://github.com/Griefed/ServerPackCreator/releases/tag/3.0.0-alpha.6).
